pool/src/pool.rs

```diff
@@ -409,6 +409,74 @@ where
 		}
 	}
 
+	/// Attempt to deaggregate a transaction and add it to the mempool
+	pub fn deaggregate_and_add_to_memory_pool(
+		&mut self,
+		tx_source: TxSource,
+		tx: transaction::Transaction,
+		stem: bool,
+	) -> Result<(), PoolError> {
+		match self.deaggregate_transaction(tx.clone()) {
+			Ok(deaggragated_tx) => self.add_to_memory_pool(tx_source, deaggragated_tx, stem),
+			Err(e) => {
+				debug!(
+					LOGGER,
+					"Could not deaggregate multi-kernel transaction: {:?}", e
+				);
+				self.add_to_memory_pool(tx_source, tx, stem)
+			}
+		}
+	}
+
+	/// Attempt to deaggregate multi-kernel transaction as much as possible based on the content
+	///of the mempool
+	pub fn deaggregate_transaction(
+		&self,
+		tx: transaction::Transaction,
+	) -> Result<Transaction, PoolError> {
+		// While the inputs outputs can be cut-through the kernel will stay intact
+		// In order to deaggregate tx we look for tx with the same kernel
+		let mut found_txs: Vec<Transaction> = vec![];
+		let mut total_kernels_found = 0;
+		// Since its a multi-kernel transaction we loop over the kernels
+		for kernel in tx.kernels.clone() {
+			// Look for tx with the same kernels
+			for (_, tx) in &self.transactions {
```

Comment:
We're looking for txs in the pool that are "subsets" of the incoming tx (based on kernel sets).

It might be simpler (and more intuitive) to put the incoming tx kernels into a set, then for each tx in the pool do the same and look at the the intersection of the two sets.

Reply:
And maybe have this as a separate fn on the pool - so we can retrieve all the txs in the pool that are candidates for deaggregating the incoming tx.

Reply:
Okay got it updating.